### PR TITLE
Issue-9: Add initial support for Validation

### LINKED ISSUE #9: Add initial support for Validation
Requires an implementation of `Tuple`.

Must include the following:

* A `Validation` type.
* The following static constructors:
    * `Failure a`
    * `Success b`
* The following static methods:
    * `concat :: Validation a b → Validation a b → Validation a b`: Concatenates two `Failure` values of `a` together, replace a `Success` with the `Failure`; otherwise, take the first `Success`.
    * `failures :: [Validation a b] → [a]`: Extracts from a list of `Validation` all of the `Failure` elements in extracted order.
    * `fromFailure :: a → Validation a b → a`: Extracts the value of a `Failure`; otherwise, returns the `defaultValue`.
    * `fromSuccess :: b → Validation a b → b`: Extracts the value of a `Success`; otherwise, returns the `defaultValue`.
    * `isFailure :: Validation a b → boolean`: Determines whether or not the argument is a `Failure`.
    * `isSuccess :: Validation a b → boolean`: Determines whether or not the argument is a `Success`.
    * `partitionValidations :: [Validation a b] → ([a], [b])`: Partitions a list of `Validation` into two lists. All `Failure` elements are extracted, in order, to the first position of the output. Similarly for the `Success` elements in the second position.
    * `successes :: [Validation a b] → [b]`: Extracts from a list of `Validation` all of the `Success` elements in extracted order.
    * `validate :: (b → boolean) → a → b → Validation a b`: Validates a value `b` and returns a `Success` of `b` if the `predicate` returns `true`; otherwise, a `Failure` of `a`.
    * `validationMap :: ([a] → c) → (b → c) → Validation a b → c`: If the value is `Failure a`, apply the first function to `[a]`; otherwise, apply the second function to `b`.

### PR DESCRIPTION
**Pull request by jlmorgan**
- [x] Squash commits before merging.

closes #9
SemVer:Minor
enhancement

### CODE REVIEW

File: docs/data/Validation/instance/equals.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# `Validation#<A, B>equals(Object other)`
+
+Determine whether or not the `other` has the same value as the current `instance`.
```
Comment (doomchild) [inline docs/data/Validation/instance/equals.md]: You have "Determine" here, but "Determines" on other predicates.

File: docs/data/Validation/static/concat.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# `Validation.<F, S>concat(Validation<F, S> second, Validation<F, S> first)`
+
+Concatenates two `Failure` values together, replace a `Success` with the `Failure`; otherwise, take the first `Success`.
```
Comment (doomchild) [inline docs/data/Validation/static/concat.md]: `replace` → `replacing`

**Review by doomchild — CHANGES_REQUESTED** [review]
Minor formatting changes needed.

File: docs/data/Validation/instance/equals.md (hunk 1 above)
Comment (jlmorgan) [reply docs/data/Validation/instance/equals.md]: Fixed.

File: docs/data/Validation/static/concat.md (hunk 2 above)
Comment (jlmorgan) [reply docs/data/Validation/static/concat.md]: Fixed.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Pull request merged**